tools/generator/templates/player/python/interface.cc.jinja2

```diff
@@ -0,0 +1,363 @@
+        throw 42;
+    PyObject* arglist = Py_BuildValue("(i)", (int) in);
+    PyObject* ret = PyObject_CallObject(enm, arglist);
+    Py_DECREF(name);
+    Py_DECREF(arglist);
+    Py_DECREF(enm);
+    return ret;
+}
+
+template <>
+{{ enum.enum_name }} python_to_cxx<PyObject*, {{ enum.enum_name }}>(PyObject* in)
+{
+  return ({{ enum.enum_name }}) python_to_cxx<PyObject*, int>(in);
+}
+{% endfor %}
+{% for struct in game.struct %}
+
+{{ struct.str_summary|cxx_comment }}
+
+template <>
+PyObject* cxx_to_python<PyObject*, {{ struct.str_name }}>({{ struct.str_name }} in)
+{
+    PyObject* tuple = PyTuple_New({{ struct.str_field|length }});
+    {% for field_name, field_type, field_comment in struct.str_field %}
+    PyTuple_SET_ITEM(tuple, {{ loop.index0 }}, ({{ field_type|cxx_to_python }}(in.{{ field_name }})));
+    {% endfor %}
+    {% if struct.str_tuple %}
+    return tuple;
+    {% else %}
+    PyObject* name = PyUnicode_FromString("{{ struct.str_name }}");
+    PyObject* cstr = PyObject_GetAttr(py_module, name);
+    Py_DECREF(name);
+    if (cstr == NULL)
+        throw 42;
+    PyObject* ret = PyObject_CallObject(cstr, tuple);
+    Py_DECREF(cstr);
+    Py_DECREF(tuple);
+    if (ret == NULL)
+        throw 42;
+    return ret;
+    {% endif %}
+}
+
+template <>
+{{ struct.str_name }} python_to_cxx<PyObject*, {{ struct.str_name }}>(PyObject* in)
+{
+    {{ struct.str_name }} out;
+    PyObject* i;
+    {% for field_name, field_type, field_comment in struct.str_field %}
+
+    {{ field_comment|cxx_comment }}
+        {% if struct.str_tuple %}
+    i = PyTuple_GetItem(in, {{ loop.index0 }});
+        {% else %}
+    i = cxx_to_python<PyObject*, int>({{ loop.index0 }});
+    i = PyObject_GetItem(in, i);
+        {% endif %}
+    if (i == NULL)
+        throw 42;
+    out.{{ field_name }} = {{ field_type|python_to_cxx }}(i);
+        {% if not struct.str_tuple %}
+    Py_DECREF(i);
+        {% endif %}
+    {% endfor %}
+
+    return out;
+}
+{% endfor %}
+
+{% for func in game.function %}
+
+{{ func.fct_summary|cxx_comment }}
+static PyObject* p_{{ func.fct_name }}(PyObject* self, PyObject* args)
+{
+    (void) self;
+    {% for arg_name, _, _ in func.fct_arg %}
+    PyObject* arg_{{ arg_name }};
+    {% endfor %}
+    if (!PyArg_ParseTuple(args, "{{ 'O' * (func.fct_arg|length) }}"
+        {%- for arg_name, _, _ in func.fct_arg -%}
+        , &arg_{{ arg_name }}
+        {%- endfor %}))
+    {
+        return NULL;
+    }
+
+    try {
+        {%+ if func is returning %}return {% endif -%}
+        {{ func.fct_ret_type|cxx_to_python }}(api_{{ func.fct_name }}(
+        {%- set comma = joiner(', ') -%}
+        {%- for arg_name, arg_type, _ in func.fct_arg -%}
+        {{ comma() }}{{ arg_type|python_to_cxx }}(arg_{{ arg_name }})
+        {%- endfor %}));
+        {% if not func is returning %}
+        Py_RETURN_NONE;
+        {% endif %}
+    } catch (...) {
+        return NULL;
+    }
+}
+{% endfor %}
+
+// API function to register.
```

Comment:
`// Register API methods.`

Reply:
No, this declares the list of functions, it doesn't register them.